### ISSUE TITLE
关于训练阶段的一些问题

### ISSUE THREAD

**Issue by ZBWpro** [kongds/Prompt-BERT#30]
首先表示感谢，作者的工作很有启发。但有一些问题需要向您请教：

1. 对于在 NLI 数据集进行有监督训练的 prompt-bert-base，你们设置的 max_sequnce_length 是 32 吗？这个值对于 NLI 数据集的部分句子而言过小，截断后可能造成语义损失。
2. 我很好奇 models.py 中，labels 为什么要定义成如下形式
  ```python
  cos_sim = cls.sim(z1.unsqueeze(1), z2.unsqueeze(0))
  
  loss_fct = nn.CrossEntropyLoss()
  labels = torch.arange(cos_sim.size(0)).long().to(input_ids.device)
  
  loss = loss_fct(cos_sim, labels)
  ```

期待您的回复，谢谢！

**Comment by kongds** [comment]
感谢关注我们的工作

1. 是的  32这个值是从SimCSE来的，具体原因可以看这个https://github.com/princeton-nlp/SimCSE/issues/112
2. 这个其实就是一个简单的对比学习损失。因为z1和z2会得到一个N乘N的相似矩阵，而矩阵对角线上的元素都是样本和正样本的相似度，所以可以用CE以[0,1,...N]的label来算这个。

**Comment by ZBWpro** [comment]
1. 感谢指点，已了解缘由。
2. 为什么 label 不是对角线全1，而其余元素全0的 one-hot 矩阵呢？

**Comment by kongds** [comment]
因为用的是CrossEntropyLoss，label应该是以label id的形式

**Comment by ZBWpro** [comment]
感谢，我会关闭这个issue
